### DISCUSSION TITLE
What should my Admin Token look like?

### DISCUSSION THREAD

**Discussion by kosymodo** [dani-garcia/vaultwarden#4813] [Q&A] [answered]
OK, so I'm having a brain fog due to the hot weather I think...

I have Vaultwarden running in Portainer on my Pi.  I initially set up the ADMIN_TOKEN as a plain text string, before hasing using the built in vaultwarden hash command.

I have manually changed my domain in the settings page, which is verified in the diagnostics tab.  Therefore, my understanding is that this has created a config.json file which overwrites the ENV settings in my container.

The config.json file shows the admin_token row as being **"admin_token": "'$argon2id$v.....'"**

When logging into the admin page, for some reason it doesn't accept the initial plain text password, instead it insists on me using the argon2 hashed string.  My understanding is that this is incorrect, right?

What am I doing wrong?

**Comment by BlackDex** [comment]
This is correct, as you mentioned your self, config.json overrides env variables, and thus everything in config.json is the source of truth for Vaultwarden.

**Reply by kosymodo** [reply]
Thanks for confirming...

So, is the format of the admin token looking correct in the config.json file?

Also, please can you confirm again that I use the hashed string as the password to log in?  And furthermore that the format of this password of sorts should be exactly the same as the text after the "admin_token": (i.e. "'$argon2id$v....'")?

**Reply by stefan0xC** [reply, marked as answer]
@kosymodo it looks like you passed the argon2id hash enquoted with `'` so it will not work, try it without the quotes and using the password you used to generate the hash should work.

**Reply by BlackDex** [reply]
Not sure what tot mean the hashed string here. It's a bit ambiguous. Did you need to enter `$argon2id$...` or the string you provided to generate that argon2id value?

Oh, wait, it looks like you add single quotes via the admin interface. I see `"'` instead of `"`.

You probably want to fix that.

**Reply by kosymodo** [reply]
Thanks both - I left it overnight and came back to it this morning.  I now have it working fine :)

For future reference (both for myself and anyone else looking at this)...

- The hashed admin token does not include the single quote at either end
- The  config.json file will include a row "admin_token": "$argon2id..." (so the hashed admin token with double quotes at either end)
- Log in using the original plain text token :)
